packages/mds-policy-service/repository/index.ts

```diff
@@ -145,6 +146,19 @@ export const PolicyRepository = ReadWriteRepository.Create('policies', { entitie
         query.andWhere('(:stop >= start_date AND (end_date IS NULL OR end_date >= :start))', active_on)
       }
 
+      if (modalities) {
+        query.andWhere(
+          "EXISTS(SELECT FROM json_array_elements(policy_json->'rules') elem WHERE elem->>'modality' = ANY(:modalities))",
+          { modalities }
+        )
+        if (modalities.includes('micromobility')) {
+          // Policies without rules or with any rule whose modality == null are micromobility by default
+          query.orWhere(
```

Comment:
I think this is all right. The modality is assumed to be `micromobility` if not explicitly stated in a rule.